homeassistant/components/modbus/sensor.py

```diff
@@ -141,10 +142,11 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        if entry.get[CONF_REGISTER_SIZE,2] * entry[CONF_COUNT] != size:
```

Comment:
you missed line 126.

Reply:
you are doing a number of entry.get in the same function, do something like mySize = entry.get... and use that.

Reply:
Line 164,167 is not corrected if register_size is 4 regs_needed would be 1